### PR TITLE
feat(bigtable/cmd/cbt): Add a timeout option

### PR DESCRIPTION
**Pull request by jimfulton**
Fixes: #2492

### CODE REVIEW

**Comment by jimfulton** [comment]
This now has a test! And includes some refactoring that makes some kinds of tests easier.

File: bigtable/cmd/cbt/testing.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+package main
+
+import (
+	"bytes"
+	"io"
+	"os"
+)
+
+func captureStdout(f func()) string {
```
Comment (garye) [inline bigtable/cmd/cbt/testing.go:25]: Document this? Wasn't immediately clear to me what f was for.

**Review by garye — APPROVED** [review]

File: bigtable/cmd/cbt/testing.go (hunk 1 above)
Comment (jimfulton) [reply bigtable/cmd/cbt/testing.go:25]: Done.

Note that here, we're mainly preventing spurious print output.  In other branches/PR, we're making assertions about captured stdout.

File: bigtable/cmd/cbt/cbt.go [hunk 2]
```diff
@@ -581,7 +607,7 @@ func doCount(ctx context.Context, args ...string) {
 	if len(args) != 1 {
 		log.Fatal("usage: cbt count <table>")
 	}
-	tbl := getClient(bigtable.ClientConfig{}).Open(args[0])
+	tbl := getTable(bigtable.ClientConfig{}, args[0])
```
Comment (crwilcox) [inline bigtable/cmd/cbt/cbt.go:610]: I am having some difficulty fully understanding the need of this refactoring and its relation to the timeout change. Can you describe it a bit?

Comment (jimfulton) [reply bigtable/cmd/cbt/cbt.go:610]: I wanted to test that a proper context (with  a timeout) was passed to `ReadRows`.  

`getTable` returns a `tableLike` object, which simply has `ReadRows`. (I think I expand it in another PR, IIRC :) )

Normally, `getTable` calls `getClient(clientConf).Open(tableName)`, but if the global `table` is set, it returns that.

`timeout_test.go` sets `table` to a mock that captures the context passed to `ReadRows` so we can assert that it was set correctly.

**Review by crwilcox — APPROVED** [review]

**Pull request merged**